appinventor/components/tests/com/google/appinventor/components/runtime/AbstractChartTest.java

```diff
@@ -0,0 +1,163 @@
+package com.google.appinventor.components.runtime;
+
+import android.graphics.drawable.ColorDrawable;
+import android.widget.RelativeLayout;
+import org.easymock.EasyMock;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+
+import static org.easymock.EasyMock.expectLastCall;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.verify;
+import static org.junit.Assert.*;
+
+/**
+ * Abstract test class for the Chart class.
+ *
+ * Contains test cases for functionality independent of the Chart type.
+ */
+public abstract class AbstractChartTest<V extends ChartView,
```

Comment:
I make use of quite a lot of abstract classes, since we have inheritance chains that rely on basically the same methods.

However, this approach might not be the best in terms of performance.

Another issue is that in the all_tests.html report, the abstract classes show up as failed tests, because they cannot be run.

Should I avoid abstract classes with Test annotations all together?